### PR TITLE
correct initialization of learning parameters for electrolysis following Adrian's paper and improve initialization procedure by2025

### PR DESCRIPTION
**Pull request by fschreyer**
# Purpose of this PR

This corrects learning parameters for electrolysis once more (follow-up to #997) and improves the initialization procedure to hit the desired capital cost values in 2025.  

See results and description of open points in this [Issue](https://github.com/remindmodel/development_issues/issues/163). 

## Type of change

(Make sure to delete from the Type-of-change list the items not relevant to your PR)

- [x] Minor change (default scenarios show only small differences)

## Checklist:

- [x] My code follows the coding etiquette
- [x] I have performed a self-review of my own code
- [x] Changes are commented, particularly in hard-to-understand areas
- [x] I have updated the in-code documentation
- [x] The model compiles and runs successfully (`Rscript start.R -q`)
- [x]  check learning initialization in test runs

## Further information (optional):

* Test runs are here: 

/p/tmp/schreyer/Modeling/remind/Current/output/Base_Learn_2022-11-18_11.51.57

/p/tmp/schreyer/Modeling/remind/Current/output/NDC_Learn_2022-11-18_16.39.53

/p/tmp/schreyer/Modeling/remind/Current/output/PkBudg500_Learn_2022-11-19_01.27.07

* Comparison of results (what changes by this PR?): 

See plots in Issue.

### CODE REVIEW

File: core/equations.gms [hunk 1]
```diff
@@ -417,7 +418,8 @@ q_limitGeopot(t,regi,peReComp(enty),rlf)..
   sum(te$teReComp2pe(enty,te,rlf), (vm_capDistr(t,regi,te,rlf) / (pm_data(regi,"luse",te)/1000)));
 
 ***  learning curve for investment costs
-q_costTeCapital(t,regi,teLearn) .. 
+***  deactivate learning for tech_stat 4 technologies before 2025 as they are not built before
+q_costTeCapital(t,regi,teLearn)$(NOT (pm_data(regi,"tech_stat",teLearn) AND t.val le 2020)) ..
```
Comment (aodenweller) [inline core/equations.gms:422]: Shouldn't it say this?
```suggestion
q_costTeCapital(t,regi,teLearn)$(NOT (pm_data(regi,"tech_stat",teLearn) eq 4 AND t.val le 2020)) .. 
```

File: core/input/generisdata_tech.prn [hunk 2]
```diff
@@ -197,14 +197,13 @@ Explanations:
...
 learning parameterization:
-Assume that 20 GW(el) electrolysis installed globally at 1200 EUR/GW(el) in 2025 
-(Ueckerdt et al. (2021), https://doi.org/10.1038/s41558-021-01032-7, Fig. S3,
- Odenweller et al. (2022), https://doi.org/10.1038/s41560-022-01097-4, Fig. 1d ).
-
-20 GW(el) * 0.65 (elh2 efficiency) * 1e-3 ~ 0.013 TW(H2) = ccap0
-1200 EUR/GW(el) / 0.65 (elh2 efficiency) * 1.1 (EUR to USD, 2015) ~ 2000 USD/kW(H2) = inco0
-Assume floor cost: 350 EUR/GW(el)
-350 EUR/GW(el) / 0.75 (long-term elh2 efficiency) * 1.1 (EUR to USD, 2015) ~ 500 USD/kW(H2) -> incolearn = 1500
+Assume that 10 GW(el) electrolysis installed globally and CAPEX of 900 EUR/kW(el) in 2025
```
Comment (aodenweller) [inline core/input/generisdata_tech.prn:200]:
```suggestion
Assume that 10 GW(el) electrolysis installed globally and CAPEX of 900 EUR/kW(el) in 2025
(slightly optimistic due to recent policies, e.g. EU Hydrogen Bank and US Inflation Reduction Act)
```

File: core/input/generisdata_tech.prn [hunk 3]
```diff
@@ -197,14 +197,13 @@ Explanations:
...
-
-20 GW(el) * 0.65 (elh2 efficiency) * 1e-3 ~ 0.013 TW(H2) = ccap0
-1200 EUR/GW(el) / 0.65 (elh2 efficiency) * 1.1 (EUR to USD, 2015) ~ 2000 USD/kW(H2) = inco0
-Assume floor cost: 350 EUR/GW(el)
-350 EUR/GW(el) / 0.75 (long-term elh2 efficiency) * 1.1 (EUR to USD, 2015) ~ 500 USD/kW(H2) -> incolearn = 1500
+Assume that 10 GW(el) electrolysis installed globally and CAPEX of 900 EUR/kW(el) in 2025 
+(Odenweller et al. (2022), https://doi.org/10.1038/s41560-022-01097-4,
+Ueckerdt et al. (2021), https://doi.org/10.1038/s41558-021-01032-7, Fig. S3).
+10 GW(el) * 0.65 (elh2 efficiency) * 1e-3 ~ 0.0065 TW(H2) = ccap0
+900 EUR/kW(el) / 0.65 (elh2 efficiency) * 1.1 (EUR to USD, 2015) ~ 1500 USD/kW(H2) = inco0
```
Comment (aodenweller) [inline core/input/generisdata_tech.prn:204]: Falko suggested 800 EUR2020/kW(el), didn't he?

**Review by aodenweller — COMMENTED** [review]
I attached some comments.

File: core/input/generisdata_tech.prn (hunk 3 above)
Comment (fschreyer) [reply core/input/generisdata_tech.prn:205]: Yeah, in the current implementation it still is a bit lower than the initialization value anyways in 2025. Gonna check when looking at the results.

**Review by aodenweller — APPROVED** [review]
Looks good.

**Comment by orichters** [comment]
Dear @schreyer, @aodenweller: Do you have a compareScenario document with runs before and after the changes you made? This PR is one of the candidates creating a drop in consumption and increased investment in 2010, 2015 that we are struggling to explain.

**Comment by aodenweller** [comment]
I'm afraid I don't as Felix did all of these runs. Does this PR increase investment into electrolysis that much in 2010 and 2015?

**Comment by orichters** [comment]
Somehow, we have dramatically reduced capital stocks. I'm not entirely sure your PR is the root cause, but at least it looks the most suspicious.

image

**Comment by aodenweller** [comment]
I'm very sorry, but I don't think I'll be of much help here. Felix did both the code changes and the runs. I only checked that the parameters in `generisdata_tech.prn` looked plausible. Have you tried reverting this PR and re-running?

**Comment by orichters** [comment]
The bug was actually totally unrelated to this PR, but found in the data REMIND was calibrated to, so my suspicion was wrong. Hopefully be fixed with [this PR in mrdrivers](https://github.com/pik-piam/mrdrivers/pull/67).

**Pull request merged**